layers/+lang/typescript/packages.el

```diff
@@ -19,6 +20,17 @@
         web-mode
         ))
 
+(unless (or (configuration-layer/layer-usedp 'javascript)
+            (configuration-layer/layer-usedp 'html))
+  (defun typescript/init-add-node-modules-path ()
+    (use-package add-node-modules-path :defer t)))
```

Comment:
Unless I'm missing something, it seems like if you use all of these layers together (or even 2 of them), nothing will define an init. I don't know if spacemacs has a way to deal w/ this (multiple layers vying to own a package) but if not, then you probably at least want to decide on a priority order for who can init it, meaning that if it's if the order is javascript->typescript->html, for example, that javascript always inits, typescript doesn't if javascript is used, and html doesn't unless neither javascript nor typescript is used.